src/webhook/mutation/pod_mutator.go

```diff
@@ -233,9 +237,13 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 		return *secretResponse
 	}
 
-	dataIngestFields, diResponse := m.ensureDataIngestSecret(ctx, ns, dkName, dk)
-	if diResponse != nil {
-		return *diResponse
+	dataIngestFields := map[string]string{}
+	if injectionInfo.enabled(DataIngest) {
+		var diResponse *admission.Response
+		dataIngestFields, diResponse = m.ensureDataIngestSecret(ctx, ns, dkName, dk)
```

Comment:
can this return an `error` instead of a `response`? the actual error response returned is always the same and could just be returned here once